packages/material-ui/src/Slide/Slide.js

```diff
@@ -161,78 +138,92 @@ class Slide extends React.Component {
-    setRef(this.props.children.ref, ref);
-  };
+  const updatePosition = React.useCallback(() => {
+    if (childrenRef.current) {
+      setTranslateValue(props, childrenRef.current);
+    }
+  }, [props]);
+
+  React.useEffect(() => {
+    const handleResize = debounce(() => {
+      // Skip configuration where the position is screen size invariant.
+      if (inProp || direction === 'down' || direction === 'right') {
+        return;
+      }
+
+      if (childrenRef.current) {
+        setTranslateValue(props, childrenRef.current);
+      }
+    }, 166); // Corresponds to 10 frames at 60 Hz.
+
+    window.addEventListener('resize', handleResize);
 
-  updatePosition() {
-    if (this.childDOMNode) {
-      setTranslateValue(this.props, this.childDOMNode);
+    return () => {
+      handleResize.clear();
+      window.removeEventListener('resize', handleResize);
+    };
+  }, [direction, inProp, props]);
+
+  React.useEffect(() => {
+    // state.mounted handle SSR, once the component is mounted, we need
+    // to properly hide it.
+    if (!inProp && !mountedRef.current) {
+      // We need to set initial translate values of transition element
+      // otherwise component will be shown when in=false.
+      updatePosition();
     }
-  }
+    mountedRef.current = true;
+  }, [inProp, updatePosition]);
 
-  render() {
-    const {
-      children,
-      direction,
-      in: inProp,
-      onEnter,
-      onEntering,
-      onExit,
-      onExited,
-      style,
-      theme,
-      ...other
-    } = this.props;
-
-    return (
-      <EventListener target="window" onResize={this.handleResize}>
-        <Transition
-          onEnter={this.handleEnter}
-          onEntering={this.handleEntering}
-          onExit={this.handleExit}
-          onExited={this.handleExited}
-          appear
-          in={inProp}
-          {...other}
-        >
-          {(state, childProps) => {
-            return React.cloneElement(children, {
-              ref: this.handleRef,
-              style: {
-                visibility: state === 'exited' && !inProp ? 'hidden' : undefined,
-                ...style,
-                ...children.props.style,
-              },
-              ...childProps,
-            });
-          }}
-        </Transition>
-      </EventListener>
-    );
-  }
+  React.useEffect(() => {
+    if (prevDirection !== direction && !inProp) {
+      // We need to update the position of the drawer when the direction change and
+      // when it's hidden.
+      updatePosition();
+    }
+    setPrevDirection(direction);
+  }, [direction, inProp, prevDirection, updatePosition]);
+
+  return (
+    <Transition
+      onEnter={handleEnter}
+      onEntering={handleEntering}
+      onExit={handleExit}
+      onExited={handleExited}
+      appear
+      in={inProp}
+      {...other}
+    >
+      {(state, childProps) => {
+        return React.cloneElement(children, {
+          ref: handleRef,
```

Comment:
`children` prop type needs the same treatment as ClickAwayListener: https://github.com/mui-org/material-ui/pull/15298/files#diff-baceb801258449c44ac6e35ee86b4fcfR100